server/src/collaborations/use-cases/findCollaborationsByStatus/find-collaborations-by-status.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  Injectable,
+  BadRequestException,
+  NotFoundException,
+} from '@nestjs/common';
+
+import { ICollaborationsRepository } from '@collaborations/interfaces';
+import { Collaboration } from '@collaborations/infra/typeorm/entities/collaboration.entity';
+import { CollaborationsStatus } from '@shared/constants';
+
+@Injectable()
+export class FindByStatus {
+  constructor(private collaborationsRepository: ICollaborationsRepository) {}
+
+  async execute(status: string): Promise<Collaboration[]> {
+    if (!status) throw new BadRequestException('Status is required');
+
+    const enumStatus = Object.keys(CollaborationsStatus).find(
+      (key) => CollaborationsStatus[key] === status.toUpperCase(),
+    );
+
+    if (!enumStatus) {
+      throw new BadRequestException('Status passed is invalid');
+    }
+    const collaborations: Collaboration[] =
+      await this.collaborationsRepository.findByStatus(status);
+
+    if (!collaborations?.length)
+      throw new NotFoundException('No collaborations found');
+
+    return collaborations;
+  }
+}
```

Comment:
É possível melhorar a legibilidade do código substituindo a lógica de validação do enumStatus por uma simples verificação se o status é uma chave válida dentro do CollaborationsStatus. Dessa forma, não seria necessária a existência da variável enumStatus dentro do código. Ficaria da seguinte forma:

`if (!(status in CollaborationsStatus)) {
      throw new BadRequestException('Status passed is invalid');
    }`

Reply:
Acredito que ao tipar o status como o enum CollaborationsStatus essa validação não será necessária